server/src/main/scala/org/scassandra/server/actors/ExecuteHandler.scala

```diff
@@ -15,115 +15,67 @@
+    implicit val protocolVersion = header.version.version
+    preparedStatement match {
+      case Some((queryText, prepared)) =>
+        val prime = primePreparedStore(queryText, execute)
+        prime.foreach(p => log.info("Found prime {}", p))
+
+        // Decode query parameters using the prepared statement metadata.
+        val dataTypes = prepared.preparedMetadata.columnSpec.map(_.dataType)
+
+        val values = extractQueryVariables(queryText, execute.parameters.values.map(_.map(_.value)), dataTypes)
+        values match {
+          case Some(v) =>
+            activityLog.recordPreparedStatementExecution(queryText, execute.parameters.consistency, v, dataTypes)
+          case None =>
+            activityLog.recordPreparedStatementExecution(queryText, execute.parameters.consistency, Nil, Nil)
         }
 
-        lazy val defaultAction = ExecuteResponse(Some(PreparedStatementExecution(preparedStatementText, executeRequest.consistency, List(), List())),
-          MessageWithDelay(msgFactory.createVoidMessage(stream)))
-        matchingPrimedAction.getOrElse(defaultAction)
-      case None =>
-        statementNotRecognised(stream, msgFactory, executeRequest)
-    }
-    action
-  }
-
-
-
-
-
-
-
-
-
-
-
 
-
-
-
-
-
-
-
-  private def statementNotRecognised(stream: Byte, msgFactory: CqlMessageFactory, executeRequest: ExecuteRequest): ExecuteResponse = {
-    val id = serializeInt(executeRequest.id)
-    val errMsg = s"Could not find prepared statement with id: ${bytes2Hex(id)}"
-    ExecuteResponse(Some(PreparedStatementExecution(errMsg, executeRequest.consistency, List(), List())),
-      MessageWithDelay(msgFactory.createErrorMessage(UnpreparedResult(errMsg, id), stream, executeRequest.consistency)))
-  }
-
-
-  private def sendMessage(msgAndDelay: MessageWithDelay, receiver: ActorRef) = {
-    msgAndDelay.delay match {
-      case None => receiver ! msgAndDelay.msg
-      case Some(duration) =>
-        log.info(s"Delaying response of prepared statement by $duration")
-        context.system.scheduler.scheduleOnce(duration, receiver, msgAndDelay.msg)(context.system.dispatcher)
-    }
-  }
-
-  private def createMessage(preparedPrime: Prime, executeRequest: ExecuteRequest ,stream: Byte, msgFactory: CqlMessageFactory, connection: ActorRef) = {
-    preparedPrime.result match {
-      case SuccessResult => msgFactory.createRowsMessage(preparedPrime, stream)
-      case result: ErrorResult => msgFactory.createErrorMessage(result, stream, executeRequest.consistency)
-      case result: FatalResult => result.produceFatalError(connection)
+        writePrime(execute, prime, header, Some(connection), alternative=Some(Reply(VoidResult)), consistency = Some(execute.parameters.consistency))
+      case None =>
+        var errMsg = s"Could not find prepared statement with id: 0x${execute.id.toHex}"
+        activityLog.recordPreparedStatementExecution(errMsg, execute.parameters.consistency, Nil, Nil)
```

Comment:
Hrmm, I may be misunderstanding, isn't the behavior the same as before?  I think the result is in this case that we record a prepared statement execution with the error message as the query text, and then an Unprepared message gets sent back to the client which will cause most drivers to repreare the statement.